pkg/infrastructure/kogito_service.go

```diff
@@ -103,3 +105,18 @@ func getServiceEndpoints(client *client.Client, namespace string, serviceHTTPRou
 	}
 	return
 }
+
+// FetchKogitoServiceURI provide kogito service URI for given instance name
+func FetchKogitoServiceURI(cli *client.Client, name, namespace string) (string, error) {
+	log.Debugf("Fetching kogito service instance URI.")
+	service := &v1.Service{}
+	if exits, err := kubernetes.ResourceC(cli).FetchWithKey(types.NamespacedName{Name: name, Namespace: namespace}, service); err != nil {
+		return "", err
+	} else if !exits {
+		return "", fmt.Errorf("service with name %s not exist for Kogito service instance in given namespace %s", name, namespace)
+	}
+	port := service.Spec.Ports[0]
+	uri := fmt.Sprintf("%s://%s:%d", port.Name, service.Name, port.TargetPort.IntVal)
```

Comment:
We should use the Kubernetes svc DNS name instead:
https://github.com/kubernetes/dns/blob/master/docs/specification.md#241---aaaaa-records

```go
// resolves to http://dataindex.mynamespace.svc.cluster.local:8080 for example
uri := fmt.Sprintf("http://%s.%s.svc.cluster.local:%d", service.Name, service.Namespace, port.TargetPort.IntVal)
```

Also `portName` can be anything, let's stick with the standard HTTP protocol since it's an internal call.

That's because we are about to change the operator to be cluster scoped instead, meaning that the operator will be in different namespace of its operands.

Reply:
done